source/plugins/go/src/oms.go

```diff
@@ -1753,17 +1791,40 @@ func InitializePlugin(pluginConfPath string, agentVersion string) {
 			Log("Routing container logs thru %s route...", ContainerLogsADXRoute)
 			fmt.Fprintf(os.Stdout, "Routing container logs thru %s route...\n", ContainerLogsADXRoute)
 		}
-	} else if strings.Compare(strings.ToLower(osType), "windows") != 0 { //for linux, oneagent will be default route
+	} else if IsWindows == false { //for linux, oneagent will be default route
 		ContainerLogsRouteV2 = true //default is mdsd route
 		Log("Routing container logs thru %s route...", ContainerLogsRoute)
 		fmt.Fprintf(os.Stdout, "Routing container logs thru %s route... \n", ContainerLogsRoute)
+	} else if IsAADMSIAuthMode { //for windows, check if MSI Auth mode, then send through AMA
+		ContainerLogsRouteV2 = true 
+		Log("Routing container logs thru %s route...", ContainerLogsV2Route)
+		fmt.Fprintf(os.Stdout, "Routing container logs thru %s route... \n", ContainerLogsV2Route)
 	}
 
+	ContainerLogSchemaVersion := strings.TrimSpace(strings.ToLower(os.Getenv("AZMON_CONTAINER_LOG_SCHEMA_VERSION")))
+	Log("AZMON_CONTAINER_LOG_SCHEMA_VERSION:%s", ContainerLogSchemaVersion)
+
+	ContainerLogSchemaV2 = false //default is v1 schema
+
+	if strings.Compare(ContainerLogSchemaVersion, ContainerLogV2SchemaVersion) == 0 && ContainerLogsRouteADX != true {
+		ContainerLogSchemaV2 = true
+		Log("Container logs schema=%s", ContainerLogV2SchemaVersion)
+		fmt.Fprintf(os.Stdout, "Container logs schema=%s... \n", ContainerLogV2SchemaVersion)
+	}
+	
 	if ContainerLogsRouteV2 == true {
-		CreateMDSDClient(ContainerLogV2, ContainerType)
+		if IsWindows {
+			if ContainerLogSchemaV2 {
+				CreateWindowsNamedPipesClient(extension.GetInstance(FLBLogger, ContainerType).GetOutputNamedPipe(ContainerLogDataType))
+			} else {
+				CreateWindowsNamedPipesClient(extension.GetInstance(FLBLogger, ContainerType).GetOutputNamedPipe(ContainerLogV2DataType))
+			}
+		} else {
+			CreateMDSDClient(ContainerLogV2, ContainerType)
+		}
 	} else if ContainerLogsRouteADX == true {
 		CreateADXClient()
-	} else { // v1 or windows
+	} else { // v1 or windows without MSI AUTH
```

Comment:
we can remove v1 from here. Direct ODS ingestion only used in case  Windows legacy auth